app/services/streaming_service_provider_data_service.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module StreamingServiceProviderDataService
+  class << self
+    def get_providers(tmdb_id:, title:, media_type:, media_format:, release_date: nil)
```

Comment:
While at first it felt a little obnoxious going from one argument to 4-5 keyword arguments, this approach makes the method much more flexible while divesting itself from any knowledge it was holding about the shape of a `movie` object. 😄  Now it just takes in the data it needs and spits out results.